test/widgets/lightbox_test.dart

```diff
@@ -271,9 +298,97 @@ void main() {
       tester.widget(find.byType(BottomAppBar));
 
       debugNetworkImageHttpClientProvider = null;
+      PaintingBinding.instance.imageCache.clear();
+    });
+
+    // FAILS
+    // testWidgets('thumbnail shown first, then main image replaces it', (tester) async {
+    //   final httpClient = FakeImageHttpClient();
+    //   debugNetworkImageHttpClientProvider = () => httpClient;
+    //   httpClient.requests[src] = FakeImageHttpClientRequest()
+    //     ..response.statusCode = HttpStatus.ok
+    //     ..response.content = kSolidBlueLargeAvatar;
+    //   httpClient.requests[thumbnail] = FakeImageHttpClientRequest()
+    //     ..response.statusCode = HttpStatus.ok
+    //     ..response.content = kSolidBlueAvatar;
+
+    //   final message = eg.streamMessage(sender: eg.otherUser);
+    //   await setupPage(
+    //     tester,
+    //     message: message,
+    //     thumbnailUrl: thumbnail,
+    //     originalWidth: 1000,
+    //     originalHeight: 1000);
+
+    //   // Both visible
+    //   tester.widget(find.byWidgetPredicate(
+    //     (widget) => widget is RealmContentNetworkImage && widget.src == thumbnail));
+    //   tester.widget(find.byWidgetPredicate(
+    //     (widget) => widget is RealmContentNetworkImage && widget.src == src));
+
+    //   print('pump');
+    //   await tester.pump();
+    //   print('done pump');
+    //   tester.widgetList<RealmContentNetworkImage>(find.byType(RealmContentNetworkImage)).forEach((w) => print(w.src));
```

Comment:
Same for this test, the frame arg in `_frameBuilder` never changes from `null`, even though `_loadingBuilder`'s `progress` become 1.0.